### PR TITLE
Jumpdest skip optimisation

### PR DESCRIPTION
**Pull request by AlexeyAkhunov**

### CODE REVIEW

**Comment by AlexeyAkhunov** [comment]
The main file to look at is `skip_analysis.go`, most of other changes are simply removing the `DestCache` in favour of skipping code bitmap generation

**Comment by holiman** [comment]
This is an interesting optimization. I wonder: do you have any numbers on how much this shaves off from processing 10M blocks?

**Comment by AskAlexSharov** [comment]
The runtime went from 39h32m to 36h40m (both times with writing down receipts)

File: core/vm/contract.go [hunk 1]
```diff
@@ -105,20 +110,6 @@ func (c *Contract) validJumpSubdest(udest uint64) bool {
 // isCode returns true if the provided PC location is an actual opcode, as
 // opposed to a data-segment following a PUSHN operation.
 func (c *Contract) isCode(udest uint64) bool {
-	// Do we have a contract hash already?
```
Comment (holiman) [inline core/vm/contract.go:108]: I think  you were also bitten by the regression fixed here: https://github.com/ethereum/go-ethereum/pull/21411 . That fix might actually have had a similar effect as this whole PR...?

Comment (mandrigin) [reply core/vm/contract.go:108]: yeah, I think we were, I remember looking at that code...

**Pull request merged**